### ISSUE TITLE
Cells executed from the input box can be executed out of order

### ISSUE THREAD

**Issue by amunger** [microsoft/vscode-jupyter#10671]
1. open a python file without having an active IW open
2. execute the cell from the file

```
# %%
print(n)
```

3. After the IW is created and while the kernel is connecting, run `n=1` from the input box

Result: the second cell added (from the input box) is executed first
![image](https://user-images.githubusercontent.com/2019016/176973631-85cee389-764e-4e8d-b602-1f4ee70a9a5f.png)

The execution count in the `CodeGenerator` is also out of sync now, so debugging and error stack traces do not behave well

**Comment by amunger** [comment]
you can also execute a set of long running cells from the file, the input box still jumps the line

```
# %%
import time
time.sleep(5)
# %%
time.sleep(5)
# %%
time.sleep(5)
#%%
time.sleep(5)
```

**Comment by rchiodo** [comment]
Yeah this makes sense. The queue is in the Jupyter Extension. The input box doesn't go through the jupyter extension. We'd need to push the queue to core or have the behavior of the input box defined by the jupyter extension I think.
